datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/PatchEntityResolver.java

```diff
@@ -0,0 +1,277 @@
+
+  private final EntityService _entityService;
+  private final EntityClient _entityClient;
+  private final EntityRegistry _entityRegistry;
+
+  @Override
+  public CompletableFuture<PatchEntityResult> get(DataFetchingEnvironment environment)
+      throws Exception {
+    final PatchEntityInput input =
+        bindArgument(environment.getArgument("input"), PatchEntityInput.class);
+    final QueryContext context = environment.getContext();
+
+    return GraphQLConcurrencyUtils.supplyAsync(
+        () -> {
+          try {
+            return patchEntity(input, context);
+          } catch (Exception e) {
+            log.error("Failed to patch entity: {}", e.getMessage(), e);
+            return new PatchEntityResult(input.getUrn(), false, e.getMessage());
+          }
+        },
+        this.getClass().getSimpleName(),
+        "patchEntity");
+  }
+
+  private PatchEntityResult patchEntity(
+      @Nonnull PatchEntityInput input, @Nonnull QueryContext context) throws Exception {
+    final Urn entityUrn = UrnUtils.getUrn(input.getUrn());
+    final Authentication authentication = context.getAuthentication();
+
+    // Check authorization
+    if (!AuthUtil.isAPIAuthorizedEntityUrns(
+        context.getOperationContext(), UPDATE, List.of(entityUrn))) {
+      throw new AuthorizationException(
+          authentication.getActor().toUrnStr() + " is unauthorized to " + UPDATE + " entities.");
+    }
+
+    // Validate aspect exists
+    final AspectSpec aspectSpec =
+        _entityRegistry
+            .getEntitySpec(entityUrn.getEntityType())
+            .getAspectSpec(input.getAspectName());
+    if (aspectSpec == null) {
+      throw new IllegalArgumentException(
+          "Aspect "
+              + input.getAspectName()
+              + " not found for entity type "
+              + entityUrn.getEntityType());
+    }
+
+    // Create patch aspect using the same pattern as OpenAPI
+    final GenericAspect patchAspect = createPatchAspect(input.getPatch());
+
+    // Create MetadataChangeProposal
+    final com.linkedin.mxe.MetadataChangeProposal mcp =
+        createMetadataChangeProposal(
+            entityUrn,
+            input.getAspectName(),
+            patchAspect,
+            input.getSystemMetadata(),
+            input.getHeaders());
+
+    // Apply the patch
+    try {
+      Urn actor = UrnUtils.getUrn(authentication.getActor().toUrnStr());
+      _entityService.ingestProposal(
+          context.getOperationContext(),
+          mcp,
+          AuditStampUtils.createAuditStamp(actor.toString()),
+          false); // synchronous for GraphQL
+
+      return new PatchEntityResult(input.getUrn(), true, null);
+    } catch (Exception e) {
+      return new PatchEntityResult(
+          input.getUrn(), false, "Failed to apply patch: " + e.getMessage());
+    }
+  }
+
+  @Nonnull
+  private GenericAspect createPatchAspect(@Nonnull List<PatchOperationInput> patchOperations) {
```

Comment:
I am seeing a lot of duplication here, between the resolvers. Lets avoid duplication when possible.